book/src/developers/designs/application_passwords.md

```diff
@@ -0,0 +1,215 @@
+# Rationale
+
+Kanidm exists to provide an authentication source for external applications.
+These applications need to have standardised ways to integrate with Kanidm to
+allow that application to interact and trust Kanidm's authentication results.
+
+For web based applications we offer Oauth2/OIDC. For Linux machines we offer a
+Kanidm specific HTTPS channel for identifying users (UNIX integration).
+Currently, for applications that don't support other protocols we offer an LDAPS
+gateway that allows users to bind using their UNIX password.
+
+However this has the issue that due to how limited Linux authentication is, UNIX
+passwords are always single factor. We also don't want the same UNIX credentials
+that are used, e.g. for sudo on machines, to be leaked and used for applications
+like email.
+
+To improve this we need a way to offer authentication services to applications
+that are unable to support anything modern.
+
+Since LDAP is the "lingua franca" of authentication and almost universally
+implemented as an authentication for all applications, we can use this to
+provide *application specific* password based authentication and remove the
+ability to bind with the UNIX password.
+
+# User experience
+
+The administrator configures two applications on their Kanidm instance. One is
+"mail" for a generic SMTP+IMAP service. The other is HTTP basic auth to a legacy
+web server. Once applications are added, the administrator adds the allowed
+users to the applications as if they were regular groups.
```

Comment:
I thought about this more and I think that for consistency, we should act the same as oauth and access controls. Rather than the application entry acting as a group, we should have the application entry nominate a single group as the "receiver" of the application, and then members of that group are allowed to access the application.  Does this seem reasonable?

Reply:
Yes, I prefer your proposal, but then users will loose the ability to inherit `memberof="applicaiton X"` (NOTE in "Application Entires" subsection)